### DISCUSSION TITLE
how can i get the return value of the global dependices?

### DISCUSSION THREAD

**Discussion by superwhys** [fastapi/fastapi#8698] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from uvicorn import run
from fastapi import FastAPI, Depends, Request


def dependence(params_a: str, params_b: str):
    return {'a': params_a, 'b': params_b}


app = FastAPI(
    docs_url="/docs",
    dependencies=[Depends(dependence)]
)


@app.get("/test")
def api_test(name: str):
    return {'name': name}


if __name__ == '__main__':
    run("run:app", host="0.0.0.0", port=9114, reload=True, debug=True)
```

### Description

I define a dependency(dependence) and use it globally, it return a dict. 
how can i get the return value of this global dependency in router 'api_test' ?

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

python3.9.7

**Comment by negadive** [comment]
you cant, since [Dependencies in path operation decorators](https://fastapi.tiangolo.com/tutorial/dependencies/dependencies-in-path-operation-decorators/#dependencies-in-path-operation-decorators) or [Global Dependencies](https://fastapi.tiangolo.com/tutorial/dependencies/global-dependencies/#global-dependencies) is introduced in the docs for dependencies that didnt return value

the way around is you store the return to [request.state](https://fastapi.tiangolo.com/tutorial/sql-databases/#about-requeststate)

**Comment by superwhys** [comment]
@negadive  Thank for the res, thats how i solved this problem. but i thought there was a better way to solve it.

**Comment by RyanTruran** [comment]
is there a way to save this in the app context instead. ie. app.state .

I have it working on a single file, but with multiple files, I run into a circular import error.

**Comment by tasercake** [comment, marked as answer]
To my understanding, FastAPI should cache the resolved results of dependency functions (for each request, I think?)

So in most cases you could do something like this without any real performance penalty:

```python
from fastapi import FastAPI, Depends

def dependence(params_a: str, params_b: str):
	# A potentially expensive operation, such as a DB call
    return {'a': params_a, 'b': params_b}

app = FastAPI(
    docs_url="/docs",
    dependencies=[Depends(dependence)]
)

@app.get("/need-dep")
def api_test(val = Depends(dependence)):
	# Do something with the dependency's (now cached) return value, 
    ...

@app.get("/no-dep")
def api_test():
	# Don't need the dependency's return value, but it gets called (and cached) anyway
    ...
```

While this doesn't really solve the issue of having to repeatedly request the dependency in every one of your view functions, this is the approach I use to enforce authentication on all of my routes, and just re-request the (hopefully cached) auth information in the view functions that need it.
